src/main/scala/io/github/scamandrill/client/MandrillClient.scala

```diff
@@ -1,445 +1,1099 @@
-    executeQuery[List[MIspInfoPool]](Endpoints.isplistpool.endpoint, marshal(isp))(unmarshal[List[MIspInfoPool]])
-  }
-
-  override def ispPoolInfo(isp: MIspPoolInfo): Future[MIspInfoPool] = {
-    executeQuery[MIspInfoPool](Endpoints.isppoolinfo.endpoint, marshal(isp))(unmarshal[MIspInfoPool])
-  }
-
-  override def ispCreatePool(isp: MIspPoolInfo): Future[MIspInfoPool] = {
-    executeQuery[MIspInfoPool](Endpoints.ispcreatep.endpoint, marshal(isp))(unmarshal[MIspInfoPool])
-  }
-
-  override def ispDeletePool(isp: MIspPoolInfo): Future[MIspDeletePoolResponse] = {
-    executeQuery[MIspDeletePoolResponse](Endpoints.ispdeletep.endpoint, marshal(isp))(unmarshal[MIspDeletePoolResponse])
-  }
-
-  override def ispSetCustomDns(isp: MIspDns): Future[MIspDnsResponse] = {
-    executeQuery[MIspDnsResponse](Endpoints.ispdns.endpoint, marshal(isp))(unmarshal[MIspDnsResponse])
+  /**
+    * Validate an API key and respond to a ping (anal JSON parser version)
+    *
+    * @return - the string "PONG!" if successful
+    */
+  def usersPing2: Future[MandrillResponse[MPingResponse]] = {
+    executeQuery[MVoid, MPingResponse](ping2, MVoid())
   }
 
-  //////////////////////////////////////////////////////////////
-  //METADATA https://mandrillapp.com/api/docs/metadata.JSON.html
-  //////////////////////////////////////////////////////////////
-
-  override def metadataList(meta: MKey): Future[List[MIMetadataResponse]] = {
-    executeQuery[List[MIMetadataResponse]](Endpoints.metalist.endpoint, marshal(meta))(unmarshal[List[MIMetadataResponse]])
+  /**
+    * Return the senders that have tried to use this account, both verified and unverified
+    *
+    * @return the senders that have tried to use this account, both verified and unverified
+    */
+  def usersSenders: Future[MandrillResponse[List[MSenderDataResponse]]] = {
+    executeQuery[MVoid, List[MSenderDataResponse]](senders, MVoid())
   }
 
-  override def metadataAdd(meta: MMeteadatapAdd): Future[MIMetadataResponse] = {
-    executeQuery[MIMetadataResponse](Endpoints.metaadd.endpoint, marshal(meta))(unmarshal[MIMetadataResponse])
-  }
+  /**
+    * Return the information about the API-connected user
+    *
+    * @return the information about the API-connected user
+    */
+  def usersInfo: Future[MandrillResponse[MInfoResponse]] = {
+    executeQuery[MVoid, MInfoResponse](info, MVoid())
+  }
+
+//  ////////////////////////////////////////////////////////////////////
+//  //MESSAGES calls https://mandrillapp.com/api/docs/messages.JSON.html
+//  ////////////////////////////////////////////////////////////////////
+//
+//  /**
+//    * Send a new transactional message through Mandrill
+//    *
+//    * @param msg - the message to send
+//    * @return - an of structs for each recipient containing the key "email" with the email address and "status" as either "sent", "queued", or "rejected"
+//    */
+//  def messagesSend(msg: MSendMessage): Future[List[MSendResponse]] = {
```

Comment:
Very much a work in progress.

I've only converted the very basic User commands from the API...everything else is commented out